nix/cabal-project.nix

```diff
@@ -0,0 +1,159 @@
+# editorconfig-checker-disable-file 
+
+{ repoRoot, inputs, lib, ... }:
+
+let
+  cabalProjectArgs = { config, pkgs, ... }: {
+
+    name = "plutus";
+
+    compiler-nix-name = lib.mkDefault "ghc927";
+
+    src = ../.;
+
+    # We would expect R to be pulled in automatically as it's a dependency of
+    # plutus-core, but it appears it is not, so we need to be explicit about
+    # the dependency on R here. Adding it as a buildInput will ensure it's
+    # added to the pkg-config env var.
+    shell = {
+      withHoogle = false;
+      buildInputs = [ pkgs.R ];
+    };
+
+    flake.variants = {
+      ghc927 = { }; # Alias for the default project
+      ghc927-profiled.modules = [{ enableProfiling = true; }];
```

Comment:
I just did some work to only write the major version everywhere, now that haskell.nix supports that as a way of writing `compiler-nix-name`, can we keep that please?